frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -639,24 +696,61 @@ function IndeterminateCheckbox({
 interface EditDocumentKeyProps {
   text: string;
   onEdit: (updatedObject: { column: string }) => void;
+  data: JobMappingFormValues[];
+  schema: string;
+  table: string;
 }
 
-function EditDocumentKey({ text, onEdit }: EditDocumentKeyProps): ReactElement {
+function EditDocumentKey(props: EditDocumentKeyProps): ReactElement {
+  const { text, onEdit, data, schema, table } = props;
   const [isEditingMapping, setIsEditingMapping] = useState<boolean>(false);
   const [inputValue, setInputValue] = useState<string>(text);
+  const [duplicateError, setDuplicateError] = useState<boolean>(false);
 
   const handleSave = () => {
     onEdit({ column: inputValue });
     setIsEditingMapping(false);
   };
 
+  const handleDocumentKeyChange = (
+    data: JobMappingFormValues[],
+    schema: string,
+    table: string,
```

Comment:
These don't really need to be passed in to this function since they are provided via the parent. Removing these from this will simplify and not shadow the parent variables.